lib/bedrock.js

```diff
@@ -211,53 +213,11 @@ api.runOnce = function(id, fn, options, callback) {
     options = {};
   }
   options = options || {};
-
-  // listen to run function once
-  process.on('message', runOnce);
-
-  const ran = false;
-  function runOnce(msg) {
-    // ignore other messages
-    if(!_isMessageType(msg, 'bedrock.runOnce') || msg.id !== id) {
-      return;
-    }
-
-    process.removeListener('bedrock.runOnce', runOnce);
-
-    if(msg.error) {
-      msg.error = errio.fromObject(msg.error, {stack: true});
-    }
-
-    if(msg.done) {
-      return callback(msg.error, ran);
-    }
-
-    const _callback = function(err) {
-      const _msg = {
-        type: 'bedrock.runOnce',
-        id,
-        done: true
-      };
-      if(err) {
-        _msg.error = errio.toObject(err, {stack: true});
-      }
-      process.send(_msg);
-    };
-
-    // run in this worker
-    if(fn.length > 0) {
-      return fn(_callback);
-    }
-    try {
-      fn();
-    } catch(e) {
-      return _callback(e);
-    }
-    _callback();
+  if(fn.length === 1) {
+    fn = promisify(fn);
   }
-
-  // schedule function to run
-  process.send({type: 'bedrock.runOnce', id, options});
+  _runOnce({type: 'bedrock.runOnce', id, fn, options})
+    .then(ran => callback(null, ran), err => callback(err));
```

Comment:
```suggestion
    .then(ran => callback(null, ran), callback);
```

Reply:
So, I considered this. I thought there might be some calling convention that might break, but maybe that's only a thing that happens in browsers these days. I'll change it and we'll see if anything goes wrong.